File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int sum =INT_MIN;
        int curlevel = 1;
        int ans = curlevel;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            int n = q.size();
            int lsum=0;
            while(n--){
                auto node = q.front();
                q.pop();
                lsum+=node->val;
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);
            }
            if(lsum>sum){
                sum=lsum;
                ans = curlevel; 
            }
            curlevel++;
        }
        return ans;
    }
};

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxLevelSum(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        msum =float('-inf')
        curlevel = 1
        ans = curlevel
        q = []
        q.append(root)
        while q:
            n = len(q)
            lsum=0
            while(n>0):
                node = q.pop(0)
                lsum+=node.val
                if(node.left):
                     q.append(node.left)
                if(node.right):
                     q.append(node.right)
                n-=1
            
            if lsum>msum:
                msum=lsum
                ans = curlevel
            
            curlevel+=1
        
        return ans